content/ideas/format-preserving-yaml.md

```diff
@@ -0,0 +1,29 @@
+---
+title: 'A format-preserving YAML library for Haskell'
+---
+
+A growing number of (Haskell) applications and libraries rely on [YAML] as a
+configuration format.  One of the motivating applications that uses YAML a lot
+is [stack].
+
+Given the number of YAML files that stack users need to deal with, it would be
+useful to be able to modify the YAML files from within Haskell.
+
+A good YAML library in Haskell [already
+exists](https://hackage.haskell.org/package/yaml).  And indeed, you can use the
+Haskell `yaml` library to load data from YAML and write it back to the disk.
+However, you'll notice that the exact formatting might not be the same.  This is
+because the datatype the library uses to represent YAML data does not include
```

Comment:
This would imply that the current `yaml` library could be extended with a better data type to support this case. That's actually _not_ the case: the entire parser and renderer is implemented as a C library taken from upstream, and it provides no location or comment preserving capabilities. It may be worth explaining that so people don't run into the yaml package dead end.

Reply:
Thanks for the feedback!  I didn't mean it that way, but I agree that my explanation was pretty vague.  I've made this more clear now.  I'm gonna go ahead and merge it so we can get it on the website sooner rather than later, but feel free to ping me if you want anything changed.